core/src/mindustry/editor/WaveInfoDialog.java

```diff
@@ -165,20 +198,25 @@ void buildGroups(){
         table.margin(10f);
 
         if(groups != null){
-            groups.sort(g -> g.begin);
+            if(sort == Sort.begin) groups.sort(g -> g.begin);
+            if(sort == Sort.totals) groups.sort(g -> g.unitAmount);
+            if(sort == Sort.health) groups.sort(g -> g.type.health);
+            if(sort == Sort.type) groups.sort(Comparator.comparing(g -> g.type));
 
             for(SpawnGroup group : groups){
                 table.table(Tex.button, t -> {
                     t.margin(0).defaults().pad(3).padLeft(5f).growX().left();
                     t.button(b -> {
                         b.left();
                         b.image(group.type.uiIcon).size(32f).padRight(3).scaling(Scaling.fit);
+                        if(group.effect != null && group.effect != StatusEffects.none) b.image(group.effect.uiIcon).size(20f).padRight(3).scaling(Scaling.fit);
                         b.add(group.type.localizedName).color(Pal.accent);
 
                         b.add().growX();
 
                         b.label(() -> (group.begin + 1) + "").color(Color.lightGray).minWidth(45f).labelAlign(Align.left).left();
 
+                        b.button(Icon.logicSmall, Styles.emptyi, () -> showEffect(group)).pad(-6).size(46f);
```

Comment:
I think this icon show display the current status (if set), instead of displaying the status by the name.

Reply:
thanks, I'll try, if it's possible